test/v200/utils/common/component_test_utils.go

```diff
@@ -0,0 +1,172 @@
+
+	LogInfoMessage("Create a volume component :")
+	component := schema.Component{}
+	component.Name = GetRandomUniqueString(8, true)
+	LogInfoMessage(fmt.Sprintf("....... Name: %s", component.Name))
+	component.Volume = &schema.VolumeComponent{}
+	devfile.componentAdded(component)
+	return component
+
+}
+
+// GetContainer returns the name of an existing, or newly created, container.
+func (devfile *TestDevfile) GetContainerName() string {
+
+	componentName := ""
+	for _, currentComponent := range devfile.SchemaDevFile.Components {
+		if currentComponent.Container != nil {
+			componentName = currentComponent.Name
+			LogInfoMessage(fmt.Sprintf("return existing container from GetContainerName  : %s", componentName))
+			break
+		}
+	}
+
+	if componentName == "" {
+		component := devfile.createContainerComponent()
+		component.Container.Image = GetRandomUniqueString(GetRandomNumber(8, 18), false)
+		componentName = component.Name
+		LogInfoMessage(fmt.Sprintf("retrun new container from GetContainerName : %s", componentName))
+	}
+
+	return componentName
+}
+
+// SetContainerComponentValues randomly sets container component attributes to random values
+func (devfile *TestDevfile) SetContainerComponentValues(component *schema.Component) {
+
+	containerComponent := component.Container
+
+	containerComponent.Image = GetRandomUniqueString(GetRandomNumber(8, 18), false)
+
+	if GetBinaryDecision() {
+		numCommands := GetRandomNumber(1, 3)
+		containerComponent.Command = make([]string, numCommands)
+		for i := 0; i < numCommands; i++ {
+			containerComponent.Command[i] = GetRandomString(GetRandomNumber(4, 16), false)
+			LogInfoMessage(fmt.Sprintf("....... command %d of %d : %s", i, numCommands, containerComponent.Command[i]))
+		}
+	}
+
+	if GetBinaryDecision() {
+		numArgs := GetRandomNumber(1, 3)
+		containerComponent.Args = make([]string, numArgs)
+		for i := 0; i < numArgs; i++ {
+			containerComponent.Args[i] = GetRandomString(GetRandomNumber(8, 18), false)
+			LogInfoMessage(fmt.Sprintf("....... arg %d of %d : %s", i, numArgs, containerComponent.Args[i]))
+		}
+	}
+
+	containerComponent.DedicatedPod = GetBinaryDecision()
+	LogInfoMessage(fmt.Sprintf("....... DedicatedPod: %t", containerComponent.DedicatedPod))
+
+	if GetBinaryDecision() {
+		containerComponent.MemoryLimit = strconv.Itoa(GetRandomNumber(4, 124)) + "M"
+		LogInfoMessage(fmt.Sprintf("....... MemoryLimit: %s", containerComponent.MemoryLimit))
+	}
+
+	if GetBinaryDecision() {
+		setMountSources := GetBinaryDecision()
+		containerComponent.MountSources = &setMountSources
+		LogInfoMessage(fmt.Sprintf("....... MountSources: %t", *containerComponent.MountSources))
+
+		if setMountSources {
+			containerComponent.SourceMapping = "/" + GetRandomString(8, false)
+			LogInfoMessage(fmt.Sprintf("....... SourceMapping: %s", containerComponent.SourceMapping))
+		}
+	}
+
+	if GetBinaryDecision() {
+		containerComponent.Env = addEnv(GetRandomNumber(1, 4))
+	} else {
```

Comment:
same as https://github.com/devfile/api/pull/348/files#r579476750

Reply:
This function is also used to update an existing exec command where, for example, the label may already be set. Will update the function description if this and other Setxxxxxx functions to make that clear.

Reply:
got it. And updating the description would help better understanding, thanks.

Reply:
Fixed in new commit